org.civicrm.reqangular/js/src/common/models/contact.js

```diff
@@ -120,16 +121,43 @@ define([
          * @return {Promise}
          */
         all: function (filters, pagination, sort, additionalParams) {
+          var response;
+          var extraFilters = {};
+
+          if (filters && filters.with_contract_in_period) {
```

Comment:
@igorpavlov the `all` method already has a mechanism to deal with "foreign filter" that seems to me you are not leveraging at all, please check what `processContactFilters` does

Reply:
Done.